### ISSUE TITLE
Mediator Metadata

### ISSUE THREAD

**Issue by Alexis-Falquier** [hyperledger/aries-framework-dotnet#161]
It seems like the mediator was intended to be able to hold metadata about a registered inbox as there is an optional parameter "metadata" on the function "CreateInboxAsync" [code](https://github.com/hyperledger/aries-framework-dotnet/blob/c2ba76fe6a2e26a97f97f72004854ea21028197f/src/Hyperledger.Aries.Routing.Edge/EdgeClientService.cs#L63) but where this function is called on the edge functionality it does not give the option to pass in metadata [code](https://github.com/hyperledger/aries-framework-dotnet/blob/c2ba76fe6a2e26a97f97f72004854ea21028197f/src/Hyperledger.Aries.Routing.Edge/EdgeProvisioningService.cs#L88) On the mediator side of things we looked at the function that handles this [code](https://github.com/hyperledger/aries-framework-dotnet/blob/c2ba76fe6a2e26a97f97f72004854ea21028197f/src/Hyperledger.Aries.Routing.Mediator/Handlers/RoutingInboxHandler.cs#L153) and found that even if the metadata was passed it does nothing with it. 

did we by chance miss something and there is still a way to set and get metadata for an inbox in the mediator? Or was this functionality abandoned?

**Comment by tmarkovski** [comment]
Yeah, this sounds like it was intended, but never actually used. Sounds like we just need to make small changes to make this work.

**Comment by Alexis-Falquier** [comment]
So we are going to need this functionality and will be happy to make these changes and make a PR do you happen to know what specifically needs to be changed? Seems like the scope of these changes should be fairly minimal correct?

**Comment by MaticDiba** [comment]
Hi @tmarkovski,
we prepared a fix for this with unit tests. Please review this #164

Any comments more then welcome.

**Comment by tmarkovski** [comment]
Thank, merged. If you need this released, can you submit a PR with the version bumped, so we can publish it to nuget?

**Comment by MaticDiba** [comment]
Thank you. We will prepare another one. Will add version bump to that one.
